src/Main/MainGui.cpp

```diff
@@ -189,6 +188,11 @@ int main(int argc, char** argv)
 #else
         App::Application::init(argc, argv);
 #endif
+        // to set window icon on wayland, the desktop file has to be available to the compositor
+        // auto desktopfile = App::Application::Config()["DesktopFileName"].c_str();
+        QGuiApplication::setDesktopFileName(
+            QString::fromLatin1(App::Application::Config()["DesktopFileName"].c_str()));
+

```

Comment:
I was unsure where was the best place to move this code IIRC, so maybe it makes more sense somewhere else, it needs to be after the branding config is loaded though. The commented code should likely be removed too.